pkg/src/lambdas/functions/crud.put-post.ts

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable */
+import AWS from 'aws-sdk'
+import LambdaRequest from '../types/lamdaRequest'
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda'
+
+import UserModel from '../../models/user.model'
+import { putSchema, postSchema } from '../schemas/crud-schema'
+
+/**
+ * fetchs lat and long coordinates by checking the given address
+ * with the given provider
+ * @param e the lambda's request object
+ */
+export const crudPutPost = async (e: APIGatewayProxyEvent, result: APIGatewayProxyResult) =>
+{
+	try
+	{
+		const dynamodb = new AWS.DynamoDB.DocumentClient({ apiVersion: '2012-08-10' })
+
+		const payload = LambdaRequest.validate(e, e.httpMethod === `POST` ? postSchema : putSchema)! as Record<string, string>
+		const user = new UserModel(payload)
+		
+		if (e.httpMethod === `POST`)
+			await user.createUser(dynamodb)
+		else
+			await user.updateUser(dynamodb)
+
+		result.statusCode = 200
+		return result
+	}
+	catch(error)
+	{
+		result.body = error.message || `Internal server error`
```

Comment:
How could you prevent internal details of the application from leaking to the outside world?

Reply:
Making sure that error is a custom exception with a user friendly error message. Any internal error message or stack should be logged, not returned to the user